### ISSUE TITLE
`brew doctor` Not Recognizing the Results of `brew update`

### ISSUE THREAD

**Issue by RandomDSdevel** [Homebrew/brew#1158]
### Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible, and retried your prior step?
- [ ] ~~If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?~~ (N/A)

### Bug report:

When I run `brew doctor -vd`, its output includes the following:  

```
⋮
Checking check_for_outdated_homebrew

Warning: Your Homebrew is outdated.
You haven't updated for at least 24 hours. This is a long time in brewland!
To update Homebrew, run `brew update`.
⋮
```

When I run _`brew update -vd`_, though, I eventually get _this_:  

```
⋮
Already up-to-date.
⋮
```

What do you guys think is going on here, and how would I fix it?  If you need any other diagnostic information to help troubleshoot this issue, I'll happily provide it.

**Comment by xinan** [comment]
I also had this issue.

Refer to issue #1151 
#1152 seems to fix that but when I do `brew update` it doesn't seem to pull the latest commits down. So I did a `brew update --force` and it went away. 

Just being curious, when I do a `brew update`, where do I pull the new version of homebrew from?

**Comment by zmwangx** [comment]
@xinan 

> when I do brew update it doesn't seem to pull the latest commits down. 

Non-developer users are now updated to tags, and locally you should see Homebrew on the "stable" branch. You need to wait for us to tag 1.0.5 before #1152 should take effect for you.

**Comment by zmwangx** [comment]
Will leave this open until #1152 lands.

**Comment by xinan** [comment]
@zmwangx Thanks!

**Comment by MikeMcQuaid** [comment]
Shipped in 1.0.5, thanks!
